src/chains/ethereum_1/templates/chainActions/erc20_approve.ts

```diff
@@ -0,0 +1,40 @@
+// import { toHex } from 'web3-utils'
+import {
+  EthereumAddress,
+  EthereumTransactionAction,
+  EthereumChainActionType,
+  EthereumDecomposeReturn,
+} from '../../models'
+import { erc20Abi } from '../abis/erc20Abi'
+
+interface erc20ApproveParams {
+  contractAddress: EthereumAddress
+  from: EthereumAddress
+  spender: EthereumAddress
+  value: number
+}
+
+export const composeAction = ({ contractAddress, from, spender, value }: erc20ApproveParams) => {
+  const contract = {
+    abi: erc20Abi,
+    parameters: [spender, value],
+    method: 'approve',
+  }
+  return {
+    to: contractAddress,
+    from,
+    contract,
+  }
+}
+
+export const decomposeAction = (action: EthereumTransactionAction): EthereumDecomposeReturn => {
```

Comment:
the return type from all decompose functions should be typed separately for each decompose function. Since we want to strongly type the return value for args
Example for Eos:
```
    const returnedData: Partial<createEscrowDefineParams> = {
      contractName: toEosEntityName(account),
      permission: toEosEntityNameOrNull(auth?.permission),
      accountName: toEosEntityName(data?.owner),
      ram: data?.ram_bytes,
...
   } 

    const partial = !returnedData?.permission
return {
      chainActionType: EosChainActionType.CreateEscrowDefine,
      args: returnedData,
      partial,
    }
}
```

Reply:
Also note in the above example the partial property now included in the return type